### PR TITLE
PLANNER-2659 Improve internal ProblemChangeDirector API

### PR DESCRIPTION
**Pull request by yurloc**
1. Fixes a bug in the examples.
2. Improves the internal problem change API so that a problem fact change can be applied easily without having to explicitly trigger variable listeners.

### JIRA
https://issues.redhat.com/browse/PLANNER-2659

### Referenced pull requests
Could replace https://github.com/kiegroup/optaplanner/pull/1821.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

**Review by rsynek — APPROVED** [review]
Nice improvement.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/common/business/SolutionBusiness.java [hunk 1]
```diff
@@ -359,8 +358,7 @@ public void doProblemChange(ProblemChange<Solution_> problemChange) {
         if (solver.isSolving()) {
             solver.addProblemChange(problemChange);
         } else {
-            problemChange.doChange(guiScoreDirector.getWorkingSolution(), problemChangeDirector);
-            guiScoreDirector.calculateScore();
+            problemChangeDirector.doProblemChange(problemChange);
```
Comment (yurloc) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/common/business/SolutionBusiness.java:361]: I wonder why is the freshly calculated score not used...

Comment (rsynek) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/common/business/SolutionBusiness.java:361]: The `SolutionBusiness` has a reference to `guiScoreDirector`, which holds the working solution. The Swing frame refreshes the score by calling the `SolutionBusiness#getScore()`

**Comment by kie-ci1** [comment]
**(tests) - optaweb-employee-rostering job** #324 was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaweb-employee-rostering/324/display/redirect or see console log:
<details>
<summary><b>Console Logs</b></summary>
 > git checkout -f fcb5ff6eed36ae77b3b601dcf3abdcaa13496768 # timeout=10<br/>Commit message: "[main] Updated LTS quarkus version (#378)"<br/> > git rev-list --no-walk fcb5ff6eed36ae77b3b601dcf3abdcaa13496768 # timeout=10<br/>Loading library jenkins-pipeline-shared-libraries@main<br/>Attempting to resolve main from remote references...<br/> > git --version # timeout=10<br/> > git --version # 'git version 1.8.3.1'<br/>using GIT_SSH to set credentials SSH: kie-ci service account with access to jboss-integration private github space<br/>[INFO] Currently running in a labeled security context<br/> > /usr/bin/chcon --type=ssh_home_t /tmp/jenkins-gitclient-ssh5010665660053281609.key<br/> > git ls-remote -h https://github.com/kiegroup/jenkins-pipeline-shared-libraries # timeout=10<br/>Found match: refs/heads/main revision a6d8872fc1ee696e346aac853dd25cae7c2749c5<br/>The recommended git tool is: NONE<br/>using credential kie-ci-user-key<br/> > git rev-parse --resolve-git-dir /var/lib/jenkins/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/optaplanner.tests.downstream.optaweb-employee-rostering/workspace@libs/jenkins-pipeline-shared-libraries/.git # timeout=10<br/>Fetching changes from the remote Git repository<br/> > git config remote.origin.url https://github.com/kiegroup/jenkins-pipeline-shared-libraries # timeout=10<br/>Fetching without tags<br/>Fetching upstream changes from https://github.com/kiegroup/jenkins-pipeline-shared-libraries<br/> > git --version # timeout=10<br/> > git --version # 'git version 1.8.3.1'<br/>using GIT_SSH to set credentials SSH: kie-ci service account with access to jboss-integration private github space<br/>[INFO] Currently running in a labeled security context<br/> > /usr/bin/chcon --type=ssh_home_t /var/lib/jenkins/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/optaplanner.tests.downstream.optaweb-employee-rostering/workspace@libs/jenkins-pipeline-shared-libraries@tmp/jenkins-gitclient-ssh3758329535554503339.key<br/> > git fetch --no-tags --progress https://github.com/kiegroup/jenkins-pipeline-shared-libraries +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>Checking out Revision a6d8872fc1ee696e346aac853dd25cae7c2749c5 (main)<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f a6d8872fc1ee696e346aac853dd25cae7c2749c5 # timeout=10<br/>Commit message: "Update logs markdown on GH to hide default (#186)"<br/> > git rev-list --no-walk a6d8872fc1ee696e346aac853dd25cae7c2749c5 # timeout=10<br/>[Pipeline] Start of Pipeline<br/>[Pipeline] node<br/>Still waiting to schedule task<br/>Waiting for next available executor on ‘kie-rhel7&&kie-mem16g&&!master’<br/>Running on rhos-01-kie-psi-rhel7-xlarge-47 in /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Checkout SCM)<br/>[Pipeline] checkout<br/>The recommended git tool is: NONE<br/>using credential kie-ci<br/>Cloning the remote Git repository<br/>Cloning repository https://github.com/kiegroup/kogito-pipelines.git<br/> > git init /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering # timeout=10<br/>Fetching upstream changes from https://github.com/kiegroup/kogito-pipelines.git<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.34.1'<br/>using GIT_ASKPASS to set credentials PSSWD: kie-ci service account with access to jboss-integration private github space<br/> > git fetch --tags --force --progress -- https://github.com/kiegroup/kogito-pipelines.git +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>Avoid second fetch<br/>Checking out Revision fcb5ff6eed36ae77b3b601dcf3abdcaa13496768 (origin/main)<br/> > git config remote.origin.url https://github.com/kiegroup/kogito-pipelines.git # timeout=10<br/> > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10<br/> > git rev-parse origin/main^{commit} # timeout=10<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f fcb5ff6eed36ae77b3b601dcf3abdcaa13496768 # timeout=10<br/>Commit message: "[main] Updated LTS quarkus version (#378)"<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] timestamps<br/>[Pipeline] {<br/>[Pipeline] timeout<br/>[2022-02-03T11:39:00.417Z] Timeout set to expire in 3 hr 0 min<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Tool Install)<br/>[Pipeline] tool<br/>[2022-02-03T11:39:00.567Z] Unpacking https://nodejs.org/dist/v12.16.3/node-v12.16.3-linux-x64.tar.gz to /home/jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/nodejs-12.16.3 on rhos-01-kie-psi-rhel7-xlarge-47<br/>[Pipeline] envVarsForTool<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Initialize)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-02-03T11:39:04.464Z] + printenv<br/>[Pipeline] archiveArtifacts<br/>[2022-02-03T11:39:04.553Z] Archiving artifacts<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space before build)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] dir<br/>[2022-02-03T11:39:05.118Z] Running in /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2022-02-03T11:39:05.145Z] [INFO] space left on the machine<br/>[Pipeline] sh<br/>[2022-02-03T11:39:05.429Z] + df -h<br/>[2022-02-03T11:39:05.429Z] Filesystem      Size  Used Avail Use% Mounted on<br/>[2022-02-03T11:39:05.429Z] devtmpfs        7.8G     0  7.8G   0% /dev<br/>[2022-02-03T11:39:05.429Z] tmpfs           7.8G     0  7.8G   0% /dev/shm<br/>[2022-02-03T11:39:05.429Z] tmpfs           7.8G   17M  7.8G   1% /run<br/>[2022-02-03T11:39:05.429Z] tmpfs           7.8G     0  7.8G   0% /sys/fs/cgroup<br/>[2022-02-03T11:39:05.429Z] /dev/vda1        80G   18G   63G  23% /<br/>[2022-02-03T11:39:05.429Z] tmpfs           1.6G     0  1.6G   0% /run/user/1000<br/>[2022-02-03T11:39:05.429Z] tmpfs           1.6G     0  1.6G   0% /run/user/0<br/>[Pipeline] echo<br/>[2022-02-03T11:39:05.447Z] [INFO] space of /home/jenkins<br/>[Pipeline] sh<br/>[2022-02-03T11:39:05.747Z] + du -h -d1 /home/jenkins<br/>[2022-02-03T11:39:05.748Z] 4.0K	/home/jenkins/.ssh<br/>[2022-02-03T11:39:05.748Z] 0	/home/jenkins/go<br/>[2022-02-03T11:39:05.748Z] 8.0K	/home/jenkins/.vnc<br/>[2022-02-03T11:39:05.748Z] 8.0K	/home/jenkins/.m2<br/>[2022-02-03T11:39:06.003Z] 2.1G	/home/jenkins/git-repos<br/>[2022-02-03T11:39:06.003Z] 8.0K	/home/jenkins/.config<br/>[2022-02-03T11:39:06.003Z] 4.0K	/home/jenkins/.cekit<br/>[2022-02-03T11:39:07.539Z] 76M	/home/jenkins/virtenvs<br/>[2022-02-03T11:39:07.539Z] 30M	/home/jenkins/remoting<br/>[2022-02-03T11:39:07.539Z] 0	/home/jenkins/.cache<br/>[2022-02-03T11:39:07.539Z] 2.1M	/home/jenkins/workspace<br/>[2022-02-03T11:39:07.539Z] 0	/home/jenkins/.pki<br/>[2022-02-03T11:39:07.539Z] 81M	/home/jenkins/tools<br/>[2022-02-03T11:39:07.539Z] 0	/home/jenkins/caches<br/>[2022-02-03T11:39:07.539Z] 2.3G	/home/jenkins<br/>[Pipeline] echo<br/>[2022-02-03T11:39:07.601Z] [INFO] space of workspace<br/>[Pipeline] sh<br/>[2022-02-03T11:39:07.929Z] + du -h -d3 /home/jenkins/workspace<br/>[2022-02-03T11:39:07.929Z] 2.1M	/home/jenkins/workspace/KIE/kogito/main<br/>[2022-02-03T11:39:07.929Z] 2.1M	/home/jenkins/workspace/KIE/kogito<br/>[2022-02-03T11:39:07.929Z] 2.1M	/home/jenkins/workspace/KIE<br/>[2022-02-03T11:39:07.929Z] 2.1M	/home/jenkins/workspace<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2022-02-03T11:39:08.270Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2022-02-03T11:39:08.358Z] [INFO] build-chain version recovered '2.3.29'<br/>[Pipeline] sh<br/>[2022-02-03T11:39:08.646Z] + npm install -g @kie/build-chain-action@2.3.29 -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2022-02-03T11:39:15.192Z] /home/jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/nodejs-12.16.3/bin/build-chain-action -> /home/jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/nodejs-12.16.3/lib/node_modules/@kie/build-chain-action/bin/build-chain-cli.js<br/>[2022-02-03T11:39:15.192Z] npm WARN notsup Unsupported engine for @kie/build-chain-action@2.3.29: wanted: {"node":">= 12.18.0"} (current: {"node":"12.16.3","npm":"6.14.4"})<br/>[2022-02-03T11:39:15.192Z] npm WARN notsup Not compatible with your version of node/npm: @kie/build-chain-action@2.3.29<br/>[2022-02-03T11:39:15.192Z] npm WARN @octokit/plugin-request-log@1.0.4 requires a peer of @octokit/core@>=3 but none is installed. You must install peer dependencies yourself.<br/>[2022-02-03T11:39:15.192Z] <br/>[2022-02-03T11:39:15.192Z] + @kie/build-chain-action@2.3.29<br/>[2022-02-03T11:39:15.192Z] added 111 packages from 89 contributors in 6.402s<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] configFileProvider<br/>[2022-02-03T11:39:16.183Z] provisioning config files...<br/>[2022-02-03T11:39:16.189Z] copy managed file [kogito_pr_settings] to file:/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering@tmp/config5333999730541234926tmp<br/>[Pipeline] {<br/>[Pipeline] withCredentials<br/>[2022-02-03T11:39:16.217Z] Masking supported pattern matches of $GITHUB_TOKEN<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-02-03T11:39:16.246Z] Warning: A secret was passed to "sh" using Groovy String interpolation, which is insecure.<br/>[2022-02-03T11:39:16.246Z] 		 Affected argument(s) used the following variable(s): [GITHUB_TOKEN]<br/>[2022-02-03T11:39:16.246Z] 		 See https://jenkins.io/redirect/groovy-string-interpolation for details.<br/>[2022-02-03T11:39:16.577Z] + build-chain-action -token=**** -df=https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml -folder=bc build pr -url=https://github.com/kiegroup/optaplanner/pull/1822 -sp=kiegroup/optaweb-employee-rostering --skipParallelCheckout -cct '(^mvn .*)||$1 -s /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering@tmp/config5333999730541234926tmp -Dmaven.wagon.http.ssl.insecure=true -Dmaven.test.failure.ignore=true'<br/>[2022-02-03T11:39:16.833Z] [INFO]  Executing pull request flow for https://github.com/kiegroup/optaplanner/pull/1822 in bc<br/>[2022-02-03T11:39:17.102Z] [INFO]  https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml exists, using it<br/>[2022-02-03T11:39:17.358Z] ::group::[PRE] Executing pre section for https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml<br/>[2022-02-03T11:39:17.359Z] [INFO]  Execute command 'export BUILD_MVN_OPTS=" -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"' in dir '/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering'<br/>[2022-02-03T11:39:17.359Z] [INFO]  Treating export command since it's not possible to run it from the runner itself<br/>[2022-02-03T11:39:17.359Z] [INFO]  The variable BUILD_MVN_OPTS has been stored with ' -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3' value<br/>[2022-02-03T11:39:17.359Z] ::notice title= [PRE] [Execution OK]::export BUILD_MVN_OPTS=" -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"<br/>[2022-02-03T11:39:17.359Z] [INFO]  Execute command 'echo "BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"' in dir '/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-employee-rostering'<br/>[2022-02-03T11:39:17.359Z] [command]/usr/bin/echo BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3<br/>[2022-02-03T11:39:17.359Z] BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3<br/>[2022-02-03T11:39:17.359Z] ::notice title= [PRE] [Execution OK]::echo "BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"<br/>[2022-02-03T11:39:17.359Z] ::endgroup::<br/>[2022-02-03T11:39:17.359Z] ::group::[Pull Request Flow] Execution Plan...<br/>[2022-02-03T11:39:17.359Z] [INFO]  ----------------------------------------------<br/>[2022-02-03T11:39:17.359Z] [INFO]  [3] projects will be executed<br/>[2022-02-03T11:39:17.359Z] <br/>[2022-02-03T11:39:17.359Z] [INFO]  [kiegroup/drools]<br/>[2022-02-03T11:39:17.359Z] [INFO]  Level Type: [upstream].<br/>[2022-02-03T11:39:17.359Z] [INFO]  mvn clean install -Dquickly ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_UPSTREAM }} ${{ env.DROOLS_BUILD_MVN_OPTS_UPSTREAM }}<br/>[2022-02-03T11:39:17.359Z] [INFO]  [ [32m'docker system prune -f'[39m ]<br/>[2022-02-03T11:39:17.359Z] <br/>[2022-02-03T11:39:17.359Z] [INFO]  [kiegroup/optaplanner]<br/>[2022-02-03T11:39:17.359Z] [INFO]  Level Type: [upstream].<br/>[2022-02-03T11:39:17.359Z] [INFO]  mvn clean install -Dquickly ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_UPSTREAM }} ${{ env.OPTAPLANNER_BUILD_MVN_OPTS_UPSTREAM }}<br/>[2022-02-03T11:39:17.359Z] [INFO]  [ [32m'docker system prune -f'[39m ]<br/>[2022-02-03T11:39:17.359Z] <br/>[2022-02-03T11:39:17.359Z] [INFO]  [kiegroup/optaweb-employee-rostering]<br/>[2022-02-03T11:39:17.359Z] [INFO]  Level Type: [current].<br/>[2022-02-03T11:39:17.359Z] [INFO]  mvn clean install -f optaweb-employee-rostering/pom.xml ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_CURRENT }} ${{ env.OPTAWEB_EMPLOYEE_ROSTERING_BUILD_MVN_OPTS }}<br/>[2022-02-03T11:39:17.359Z] [INFO]  [ [32m'docker system prune -f'[39m ]<br/>[2022-02-03T11:39:17.359Z] <br/>[2022-02-03T11:39:17.359Z] [INFO]  ----------------------------------------------<br/>[2022-02-03T11:39:17.359Z] ::endgroup::<br/>[2022-02-03T11:39:17.359Z] ::group::[Pull Request Flow] Checking out kiegroup/optaplanner and its dependencies<br/>[2022-02-03T11:39:17.359Z] [INFO]  Tree for project kiegroup/optaweb-employee-rostering. Dependencies: <br/>[2022-02-03T11:39:17.359Z] kiegroup/drools,<br/>[2022-02-03T11:39:17.359Z] kiegroup/optaplanner,<br/>[2022-02-03T11:39:17.359Z] kiegroup/optaweb-employee-rostering<br/>[2022-02-03T11:39:17.359Z] [INFO]  [kiegroup/drools] Checking out project<br/>[2022-02-03T11:39:17.616Z] [WARN]  project github.com/yurloc/drools:planner-2659 does not exist. It's not necessarily an error.<br/>[2022-02-03T11:39:17.616Z] [WARN]  project github.com/kiegroup/drools:planner-2659 does not exist. It's not necessarily an error.<br/>[2022-02-03T11:39:17.876Z] [INFO]  [kiegroup/drools] Checking out 'https://github.com/kiegroup/drools:main' into 'bc/kiegroup_drools'<br/>[2022-02-03T11:40:04.566Z] [INFO]  [kiegroup/drools] Checked out.<br/>[2022-02-03T11:40:04.566Z] [INFO]  [kiegroup/drools] Clonning folder bc/kiegroup_drools into drools<br/>[2022-02-03T11:40:07.084Z] [INFO]  [kiegroup/drools] Clonned folder bc/kiegroup_drools into bc/kiegroup_drools/drools<br/>[2022-02-03T11:40:07.084Z] [INFO]  [kiegroup/optaplanner] Checking out project<br/>[2022-02-03T11:40:07.649Z] [INFO]  [kiegroup/optaplanner] Merging https://github.com/kiegroup/optaplanner:main into https://github.com/yurloc/optaplanner:planner-2659<br/>[2022-02-03T11:41:29.847Z] [INFO]  [kiegroup/optaplanner] Checked out.<br/>[2022-02-03T11:41:29.847Z] [INFO]  [kiegroup/optaplanner] Clonning folder bc/kiegroup_optaplanner into optaplanner<br/>[2022-02-03T11:41:51.849Z] [INFO]  [kiegroup/optaplanner] Clonned folder bc/kiegroup_optaplanner into bc/kiegroup_optaplanner/optaplanner<br/>[2022-02-03T11:41:51.849Z] [INFO]  [kiegroup/optaweb-employee-rostering] Checking out project<br/>[2022-02-03T11:41:51.849Z] [WARN]  yurloc/optaweb-employee-rostering not found. Trying to get it by forked project list.<br/>[2022-02-03T11:41:51.849Z] [ERROR]  You have exceeded a secondary rate limit. Please wait a few minutes before you try again.<br/>[Pipeline] }<br/>[Pipeline] // withCredentials<br/>[Pipeline] }<br/>[2022-02-03T11:41:52.044Z] Deleting 1 temporary files<br/>[Pipeline] // configFileProvider<br/>[Pipeline] }<br/>[Pipeline] // script<br/>Post stage<br/>[Pipeline] junit<br/>[2022-02-03T11:41:52.259Z] Recording test results<br/>[2022-02-03T11:41:52.967Z] No test report files were found. Configuration error?<br/>Error when executing always post condition:<br/>Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to rhos-01-kie-psi-rhel7-xlarge-47<br/>		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1797)<br/>		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)<br/>		at hudson.remoting.Channel.call(Channel.java:1001)<br/>		at hudson.FilePath.act(FilePath.java:1166)<br/>		at hudson.FilePath.act(FilePath.java:1155)<br/>		at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:107)<br/>		at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:153)<br/>		at hudson.tasks.junit.JUnitResultArchiver.parseAndSummarize(JUnitResultArchiver.java:247)<br/>		at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:63)<br/>		at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:29)<br/>		at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)<br/>		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)<br/>hudson.AbortException: No test report files were found. Configuration error?<br/>	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:154)<br/>	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3338)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:211)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:54)<br/>	at hudson.remoting.Request$2.run(Request.java:376)<br/>	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)<br/>	at java.util.concurrent.FutureTask.run(FutureTask.java:266)<br/>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)<br/>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)<br/>	at java.lang.Thread.run(Thread.java:748)<br/><br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-02-03T11:41:53.343Z] + rm -rf console.log<br/>[Pipeline] sh<br/>
</details>

**Comment by yurloc** [comment]
Jenkins run optaweb-employee-rostering tests.

**Pull request merged**